app/Console/Commands/DispatchUserQuizClosedEvent.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Console\Commands;
+
+use App\Jobs\CloseUserQuizJob;
+use App\Models\Quiz;
+use Illuminate\Console\Command;
+use Illuminate\Database\Eloquent\Collection;
+
+class DispatchUserQuizClosedEvent extends Command
+{
+    protected $signature = "app:dispatch-user-quiz-closed-event";
+    protected $description = "Dispatches the UserQuizClosed event for quizzes that are closing today.";
+
+    public function handle(): void
+    {
+        foreach ($this->getClosingTodayQuizzes() as $quiz) {
+            CloseUserQuizJob::dispatch($quiz)->delay($quiz->closeAt);
+        }
+    }
+
+    /**
+     * @return Collection<Quiz>
+     */
+    public function getClosingTodayQuizzes(): Collection
+    {
+        return Quiz::query()
+            ->whereNotNull("locked_at")
+            ->whereRaw("(scheduled_at + INTERVAL '1 minute' * duration) >= ?", [now()])
+            ->whereRaw("DATE(scheduled_at + INTERVAL '1 minute' * duration) = ?", [today()])
+            ->get();
```

Comment:
I'm not a big fan of using raw SQL :eyes:

Reply:
It seems to me that Eloquent doesn't provide a way to add a value expressed in minutes to a timestamp. At least, I couldn't find anything like that in the documentation. That's why I had to use raw SQL here. I can revert it to the previous, less optimal version where I fetch all the quizzes from the database and then filter them using PHP. In the end, it shouldn't significantly affect performance since this command runs only once a day anyway.

Reply:
I see. So maybe let's stay with that? :eyes:

Reply:
done